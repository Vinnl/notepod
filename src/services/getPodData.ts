import { describeSubject, VirtualDocument, describeDocument, describeContainer, Reference } from 'plandoc';
import { space, solid, rdf, schema } from 'rdf-namespaces';

export type PodData = {
  webId: Reference,
  profileDoc: VirtualDocument,
  publicTypeIndex: VirtualDocument,
  notesDoc: VirtualDocument,
};

export function getPodData(webId: string): PodData {
  const profileDoc = describeDocument().isFoundAt(webId);
  const profile = describeSubject().isFoundAt(webId);

  const storage = describeContainer().isFoundOn(profile, space.storage);

  const publicTypeIndex = describeDocument().isFoundOn(profile, solid.publicTypeIndex)

  const notesTypeRegistration = describeSubject()
    .isEnsuredIn(publicTypeIndex)
    .withRef(rdf.type, solid.TypeRegistration)
    .withRef(solid.forClass, schema.TextDigitalDocument);

  const notesDoc = describeDocument()
    .isEnsuredOn(notesTypeRegistration, solid.instance, storage);

  return {
    webId,
    profileDoc,
    publicTypeIndex,
    notesDoc,
  };
}
