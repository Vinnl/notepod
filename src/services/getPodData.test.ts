import { experimental_serialise } from 'plandoc';
import { getPodData} from './getPodData';

describe('getPodData', () => {
  it('should provide the route to the Profile Document', () => {
    expect(experimental_serialise(getPodData('https://arbitrary.pod/profile.ttl#webid').profileDoc))
      .toMatchSnapshot();
  });

  it('should provide the route to the Public Type Index', () => {
    expect(experimental_serialise(getPodData('https://arbitrary.pod/profile.ttl#webid').publicTypeIndex))
      .toMatchSnapshot();
  });

  it('should provide the route to the Notes Document', () => {
    expect(experimental_serialise(getPodData('https://arbitrary.pod/profile.ttl#webid').notesDoc))
      .toMatchSnapshot();
  });

  it('should provide the given WebID', () => {
    expect(getPodData('https://arbitrary.pod/profile.ttl#webid').webId)
      .toBe('https://arbitrary.pod/profile.ttl#webid');
  });
});
