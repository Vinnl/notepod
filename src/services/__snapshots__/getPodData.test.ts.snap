// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPodData should provide the route to the Notes Document 1`] = `
Object {
  "acl": Object {},
  "fallbackContainer": Object {
    "predicate": "http://www.w3.org/ns/pim/space#storage",
    "subject": Object {
      "reference": "https://arbitrary.pod/profile.ttl#webid",
      "type": "IsFoundAt",
    },
    "type": "IsFoundOn",
  },
  "predicate": "http://www.w3.org/ns/solid/terms#instance",
  "subject": Object {
    "document": Object {
      "predicate": "http://www.w3.org/ns/solid/terms#publicTypeIndex",
      "subject": Object {
        "reference": "https://arbitrary.pod/profile.ttl#webid",
        "type": "IsFoundAt",
      },
      "type": "IsFoundOn",
    },
    "locator": Object {
      "references": Array [
        Object {
          "object": "http://www.w3.org/ns/solid/terms#TypeRegistration",
          "predicate": "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
        },
        Object {
          "object": "http://schema.org/TextDigitalDocument",
          "predicate": "http://www.w3.org/ns/solid/terms#forClass",
        },
      ],
    },
    "type": "IsEnsuredIn",
  },
  "type": "IsEnsuredOn",
}
`;

exports[`getPodData should provide the route to the Profile Document 1`] = `
Object {
  "reference": "https://arbitrary.pod/profile.ttl#webid",
  "type": "IsFoundAt",
}
`;

exports[`getPodData should provide the route to the Public Type Index 1`] = `
Object {
  "predicate": "http://www.w3.org/ns/solid/terms#publicTypeIndex",
  "subject": Object {
    "reference": "https://arbitrary.pod/profile.ttl#webid",
    "type": "IsFoundAt",
  },
  "type": "IsFoundOn",
}
`;
